### PR TITLE
updated most dependencies

### PR DESCRIPTION
**Pull request by trytomakeyouprivate**
These dependencies where nearly all outdated!

I tried to always use the latest (stable) tag, but sometimes the tags where years old, so latest commit was taken.

also note that some repos are now archived.

```
https://gitlab.com/yawning/bsaes
# has no tags, latest commit

https://github.com/pion/udp
# is archived!

https://github.com/pion/logging
# last tag 2019, using latest commit

https://github.com/pion/datachannel
# last tag 2022, using latest commit

https://github.com/pion/ice
# last tag 2019, using latest commit

https://github.com/pion/turn
# last tag 2018! using latest commit

https://github.com/pion/sdp
# last tag 2020! using latest commit

https://github.com/tjfoc/gmsm
# last tag 2021, using latest commit 2022

https://github.com/xtaci/smux
# last tag Feb, using latest commit seems small

https://github.com/klauspost/reedsolomon
# last tag June, using latest commit

https://github.com/pkg/errors
# is archived!

https://github.com/onionshare/onionshare.git
# latest tag 2.6 a year old?

https://files.pythonhosted.org/packages/f6/f0/10642828a8dfb741e5f3fbaac830550a518a775c7fff6f04a007259b0548/py-1.11.0-py2.py3-none-any.whl #  could not find this package

should something newer than pysocks 1.7.1 be used?
```

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

## needed fixes
Line 24-49 are copied from https://github.com/onionshare/onionshare and meant for local build. They need to be updated for this repo

### CODE REVIEW

**Comment by trytomakeyouprivate** [comment]
removed xdg-config access as the build was failing

File: org.onionshare.OnionShare.yaml [hunk 1]
```diff
@@ -2,11 +2,11 @@
 app-id: org.onionshare.OnionShare
 command: onionshare
 runtime: org.kde.Platform
-runtime-version: "5.15-21.08"
+runtime-version: "6.5"
```
Comment (hfiguiere) [inline org.onionshare.OnionShare.yaml:5]: why 6.5?

1. latest for Qt6 is 6.6
2. latest for Qt5 is 5.15-23.08

the latter is probably what you want.

Comment (trytomakeyouprivate) [reply org.onionshare.OnionShare.yaml]: thanks, Qt5 is probably what we should use? There need to be some adaptions to work on Qt6 probably, right?

File: org.onionshare.OnionShare.yaml [hunk 2]
```diff
@@ -21,31 +21,37 @@ cleanup:
...
-      - /bin
-    sources:
-      - type: archive
-        sha256: 7ff5f1cc4291fffb6d5a3098b3090abe4d415da2adec740b4e901893d95d7137
-        url: https://download.qt.io/official_releases/QtForPython/pyside2/PySide2-5.15.3-src/pyside-setup-opensource-src-5.15.3.tar.xz
-      - type: shell
-        commands:
-          - mkdir -p /app/include/qt5tmp && cp -R /usr/include/Qt* /app/include/qt5tmp # https://bugreports.qt.io/browse/PYSIDE-787
-          - sed -i 's|\(--include-paths=\)|\1/app/include/qt5tmp:|' sources/pyside2/cmake/Macros/PySideModules.cmake
+  - name: pyside6
```
Comment (hfiguiere) [inline org.onionshare.OnionShare.yaml:24]: on qt5 ?

File: org.onionshare.OnionShare.yaml [hunk 3]
```diff
@@ -21,31 +21,37 @@ cleanup:
...
+      - name: pyside6-essentials
+        only-arches:
+          - x86_64
+        buildsystem: simple
+        build-commands:
+          - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+            --prefix=${FLATPAK_DEST} "pyside6-essentials" --no-build-isolation
+        sources:
+          - type: file
+            url: https://files.pythonhosted.org/packages/d0/de/9a089e91c2e0fe4f122218bba4f9dbde46338659f412739bd9db1ed9df4f/PySide6_Essentials-6.5.2-cp37-abi3-manylinux_2_28_x86_64.whl
```
Comment (hfiguiere) [inline org.onionshare.OnionShare.yaml:37]: how will this work on aarch64?

**Pull request closed without merging**